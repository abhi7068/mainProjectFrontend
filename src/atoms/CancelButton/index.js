import React from 'react';

import Button from '@material-ui/core/Button';
import { makeStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
const useStyles = makeStyles((theme) => ({
  root: {
    '&:hover': {
      background: 'transparent',
      transition: 'none',
    },
    '&:active': {
      transition: 'none',
      boxShadow: 'none',
      background: 'transparent',
    },
  },
  cancelButton: {
    background: 'transparent',
    border: 'none',
    cursor: 'pointer',
    color: '#fff',
    fontSize: '32px',
    lineHeight: '1',
    verticalAlign: 'top',
    position: 'absolute',
    top: '-5px',
    right: '-50px',
    '&hover': {
      cursor: 'pointer',
    },
  },
}));

function CancelButton({ children, ...rest }) {
  const classes = useStyles();
  return (
    <Button
      {...rest}
      className={classes.cancelButton}
      classes={{ root: classes.root }}
    >
      {children}
    </Button>
  );
}
CancelButton.propTypes = {
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
};

export default CancelButton;
