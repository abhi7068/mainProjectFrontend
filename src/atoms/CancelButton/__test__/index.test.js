import React from 'react';
import CancelButton from '../index';
import { render, cleanup } from '@testing-library/react';
afterEach(cleanup);
it('it renders', () => {
  render(<CancelButton />);
});
it('snapshot', () => {
  const { asFragment } = render(<CancelButton />);
  expect(asFragment()).toMatchSnapshot();
});
