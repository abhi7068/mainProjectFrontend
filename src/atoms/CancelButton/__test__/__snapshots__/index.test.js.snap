// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root makeStyles-root-45 MuiButton-text makeStyles-cancelButton-46"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    />
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
